### PR TITLE
Minor improvements for arrow_filter_policy

### PR DESCRIPTION
**Pull request by mhaseeb123**
This PR make the hasher to `arrow_filter_policy` a template type for users to simply pass the hasher without specifying the `Key` type twice. See this comment for discussion: https://github.com/NVIDIA/cuCollections/issues/653#issuecomment-2530111406. Also update member types `argument_type` and `hash_result_type` to directly use `Key` and `std::uint64_t` for this policy.

### CODE REVIEW

File: include/cuco/detail/bloom_filter/arrow_filter_policy.cuh [hunk 1]
```diff
@@ -81,10 +81,10 @@ namespace cuco::detail {
...
  */
-template <class Key, class XXHash64>
+template <class Key, template <typename> class XXHash64>
 class arrow_filter_policy {
  public:
-  using hasher             = XXHash64;       ///< 64-bit XXHash hasher for Arrow bloom filter policy
+  using hasher             = XXHash64<Key>;  ///< 64-bit XXHash hasher for Arrow bloom filter policy
   using word_type          = std::uint32_t;  ///< uint32_t for Arrow bloom filter policy
   using hash_argument_type = typename hasher::argument_type;  ///< Hash function input type
   using hash_result_type   = decltype(std::declval<hasher>()(
```
Comment (PointKernel) [inline include/cuco/detail/bloom_filter/arrow_filter_policy.cuh]:
```suggestion
  using key_type = Key;  ///< Hash function input type
  using hash_value_type   = uint64_t;
```
We could still keep the current definition as it is and adding a build time check verifying it's the same as `uint64_t`.

Comment (mhaseeb123) [reply include/cuco/detail/bloom_filter/arrow_filter_policy.cuh]: With this, we can just revert the change in libcudf's `XXHash64` and I will update the `arrow_fitler_policy.cuh` that I copied over to libcudf as well (which will be deleted with the cuco bump anyway)

File: include/cuco/detail/bloom_filter/arrow_filter_policy.cuh [hunk 2]
```diff
@@ -81,14 +80,13 @@ namespace cuco::detail {
...
+template <class Key, template <typename> class XXHash64>
 class arrow_filter_policy {
  public:
-  using hasher             = XXHash64;       ///< 64-bit XXHash hasher for Arrow bloom filter policy
+  using hasher             = XXHash64<Key>;  ///< 64-bit XXHash hasher for Arrow bloom filter policy
   using word_type          = std::uint32_t;  ///< uint32_t for Arrow bloom filter policy
-  using hash_argument_type = typename hasher::argument_type;  ///< Hash function input type
-  using hash_result_type   = decltype(std::declval<hasher>()(
-    std::declval<hash_argument_type>()));  ///< hash function output type
+  using hash_argument_type = Key;            ///< Hash function input type
```
Comment (mhaseeb123) [inline include/cuco/detail/bloom_filter/arrow_filter_policy.cuh]: ~Kept the names of these member types consistent with the rest of the file as well as the members of `default_filter_policy`~ Nvm, let's update here at least

**Review by PointKernel — APPROVED** [review]
Thanks for the quick fix @mhaseeb123

**Pull request merged**